### ISSUE TITLE
high nfs overhead

### ISSUE THREAD

**Issue by nick-youngblut** [bbuchfink/diamond#283]
This is assuming an nfs:

If many instances of diamond are run in parallel (eg., many cluster jobs), the number of GETATTR rpc calls can increase to the many 100s of 1000s, which can greatly slow down the file system. In the diamond temp_dir, diamond creates a bunch of files labeled like ".nfs0000002e015a544300023767" (100s of files). It seems that diamond is either calling stat(2) on these files constantly or multiple nfs clients have these files open at the same time. 

In other words, on our nfs, it appears that diamond does not scale due to the high number of GETATTR rpc calls that it generates. At least, that is with the following settings: `--sensitive --max-target-seqs 20 --block-size 3 --index-chunks 2`. Is there any way to reduce the number of GETATTR rpc calls?  I don't see anything in the docs about this.

**Comment by bbuchfink** [comment]
Thanks, I will have to look into that but it probably won't be that easy to fix. For now I recommend setting the temp directory to a fast local drive or simply `/dev/shm`, that is the best option for high memory nodes. You can use the same temp directory for parallel runs.

**Comment by nick-youngblut** [comment]
Thanks for the advice. Sadly, it seems that using `/dev/shm` doesn't help. The GETATTR rpc calls are still really high when running diamond.

**Comment by bbuchfink** [comment]
Ok, I will definitely have to look into that. Hopefully I will be able to provide a fix in the next days.

**Comment by tseemann** [comment]
@nick-youngblut does your clisyer/HPC have a special fast location for temp or scratch data? Often it is `/scratch`.  Also, do you happen to have `$TMPDIR` set?

**Comment by nick-youngblut** [comment]
No, we do not have a special fast location for temp/scratch data. I currently have diamond set to `--tmpdir /dev/shm/diamond_tmp/`

**Comment by bbuchfink** [comment]
Please try to use the latest github commit. This version now creates a lot less temporary files. Let me know if this improves your problem.
